StrongInject.Generator/Visitors/DependencyCheckerVisitor.cs

```diff
@@ -146,7 +146,7 @@ protected override bool ShouldVisitAfterUpdateState(InstanceSource source, State
         
         protected override void AfterVisit(InstanceSource source, State state)
         {
-            if (source is { IsAsync: true, Scope: not Scope.SingleInstance } and not DelegateSource && !state.IsScopeAsync
+            if (source is { IsAsync: true, Scope: not Scope.SingleInstance } and not (DelegateSource or OwnedSource) && !state.IsScopeAsync
```

Comment:
It might have been a mistake of me to have delegates set IsAsync to true when they return a `Task<T>`. and not to use a separate property since as you can see here it means rather different things for a Delegate and an Owned compared to anything else. Still this is good for now, and I can consider the refactoring later.